ledger/src/blockstore_db.rs

```diff
@@ -1468,6 +1468,31 @@ pub struct WriteBatch {
     write_batch: RWriteBatch,
 }
 
+pub struct BlockstoreByteReference<'a> {
+    slice: DBPinnableSlice<'a>,
+}
+
+impl<'a> From<DBPinnableSlice<'a>> for BlockstoreByteReference<'a> {
+    fn from(slice: DBPinnableSlice<'a>) -> Self {
+        Self { slice }
+    }
+}
+
+impl std::ops::Deref for BlockstoreByteReference<'_> {
+    type Target = [u8];
+
+    fn deref(&self) -> &[u8] {
```

Comment:
```rust
-> Self::Target
```
I would add `#[inline]` too.

Reply:
done in https://github.com/anza-xyz/agave/pull/4674/commits/70bcaf74f009d5617aa55a794d55babc16fd68f1